portfolio/src/main/java/com/google/sps/servlets/NumCommentsServlet.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+import com.google.sps.data.Comment;
+
+/** 
+ * Returns total number of comments currently in the system 
+ * TODO for when comments have a DELETED = T/F
+ * Apply a filter to the new Query...
+ */
+@WebServlet("/num-comments")
+public class NumCommentsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query("Comment");
```

Comment:
Write a query to select count(*), that should be faster than just selecting everything and counting from there, which is what it looks like you're doing here (granted I am not totally familiar with the API but I'm assuming datastore.prepare(query) executes the query).

Reply:
https://stackoverflow.com/questions/4979358/app-engine-datastore-viewer-how-to-show-count-of-records-using-gql

It looks like there is not a straightforward way to do this sadly, at least not in a dev environment. I agree this is a slow way, and in the future i will probably modify the datastore to keep a built in counter that i can reference.